src/pages/login/login.ts

```diff
@@ -130,23 +149,32 @@ export class LoginPage extends BasePageComponent {
   }
 
   isInternetConnectionError = (): boolean => {
-    return !this.hasUserLoggedOut && this.authenticationError === AuthenticationError.NO_INTERNET;
+    return !this.hasUserLoggedOut && this.appInitError === AuthenticationError.NO_INTERNET;
   }
 
   isUserCancelledError = (): boolean => {
-    return !this.hasUserLoggedOut && this.authenticationError === AuthenticationError.USER_CANCELLED;
+    return !this.hasUserLoggedOut && this.appInitError === AuthenticationError.USER_CANCELLED;
   }
 
   isUnknownError = (): boolean => {
     return !this.hasUserLoggedOut &&
-      this.authenticationError &&
-      this.authenticationError.valueOf() !== AuthenticationError.USER_CANCELLED &&
-      this.authenticationError.valueOf() !== AuthenticationError.NO_INTERNET &&
-      this.authenticationError.valueOf() !== AuthenticationError.USER_NOT_AUTHORISED;
+      this.appInitError &&
+      this.appInitError.valueOf() !== AuthenticationError.USER_CANCELLED &&
+      this.appInitError.valueOf() !== AuthenticationError.NO_INTERNET &&
+      this.appInitError.valueOf() !== AuthenticationError.USER_NOT_AUTHORISED;
   }
 
   isUserNotAuthorised = (): boolean => {
-    return !this.hasUserLoggedOut && this.authenticationError === AuthenticationError.USER_NOT_AUTHORISED;
+    return !this.hasUserLoggedOut && this.appInitError === AuthenticationError.USER_NOT_AUTHORISED;
+  }
+
+  showErrorDetails() {
+    const alert = this.alertCtrl.create({
+      title: 'Error details',
+      subTitle: JSON.stringify(this.appInitError),
```

Comment:
you could also use `message` instead of `subTitle` and your error text would be bigger, maybe easier to read.

Reply:
Will try it out :)

Reply:
Think we might need to keep it quite small to allow for a large error object with stack trace etc.